core/src/main/java/org/infinispan/remoting/transport/impl/SingleTargetRequest.java

```diff
@@ -26,52 +26,46 @@ public class SingleTargetRequest<T> extends AbstractRequest<T> {
    private static final Log log = LogFactory.getLog(SingleTargetRequest.class);
 
    // Only changes from non-null to null
-   @GuardedBy("responseCollector")
-   private RequestTracker requestTracker;
+   private final AtomicReference<RequestTracker> requestTracker;
```

Comment:
You can remove `AtomicReference` and make the original field final. I don't remember the reason why I'm setting it to null, and It just complicates the code for nothing.

Reply:
Keeping it protects from concurrent calls. This prevents mismatch between metrics and the actual operation performed as well.

Reply:
You can change the return value of the `RequestTacker` methods to a boolean to signal if the response is accepted or not. 
Although, it doesn't seem a big issue.

Reply:
It would, but this class is much simpler (single response) and doesn't need the fancy exclusive running stuff.